### PR TITLE
Updated upload/download artifact to v4 - Changed test reporting action

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Fixed log artifact naming to prevent clobbering logs for separate policy packages
* Upgraded upload/download artifacts to v4 to remove nodejs 16 warnings. As a result, had to change dorny-test-reporter as it did not support GitHub v4 artifacts.
* Added EnricoMi/publish-unit-test-result-action for test reporting

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by MariusNi — DISMISSED** [review]
:shipit:

**Review by robertschaedler3 — APPROVED** [review]

**Review by simonjaeger — APPROVED** [review]

**Review by MariusNi — DISMISSED** [review]
:shipit:

File: .github/workflows/universalnrp-test-run.yml [hunk 1]
```diff
@@ -115,6 +125,8 @@ jobs:
           EOL
 
           sudo LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/omi/lib/ pwsh -Command $script
+          ls -l
+          sudo chmod a+r *testResults.xml
```
Comment (MariusNi) [inline .github/workflows/universalnrp-test-run.yml:129]: Thought that sudo was failing on copying the NRP log, not the test report XML. Where's that one?

**Review by MariusNi — DISMISSED** [review]
:shipit:

**Review by MariusNi — DISMISSED** [review]
:shipit:

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**